code/game/machinery/keypads.dm

```diff
@@ -0,0 +1,237 @@
+//Wallframes
+/obj/item/wallframe/keypad
+	name = "keypad frame"
+	desc = "Used for building keypads."
+	icon_state = "keypad_frame"
+	result_path = /obj/machinery/keypad
+	materials = list(MAT_METAL=MINERAL_MATERIAL_AMOUNT)
+	var/password = ""
+
+/obj/item/wallframe/keypad/attack_self(mob/user)
+	if(!user)
+		return 0
+	if(!password)
+		var/pass = input(user,"Please input a new password for this keypad.","Set Password") as null|num
```

Comment:
You forgot to verify if the `mob/user` is still near the `src` after `input()` completes